### PR TITLE
Fix assorted issues with logscursor

### PR DESCRIPTION
**Pull request by joshuacolvin0**

### CODE REVIEW

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 1]
```diff
@@ -1735,22 +1743,34 @@ bool ArbCore::handleLogsCursorRequested(ReadTransaction& tx,
...
+        return;
     }
 
     auto current_count_result =
         logsCursorGetCurrentTotalCount(tx, cursor_index);
     if (!current_count_result.status.ok()) {
         std::cerr << "Unable to get logs cursor current total count: "
-                  << cursor_index << "\n";
-        return false;
+                  << cursor_index << std::endl;
```
Comment (hkalodner) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1757]: Should set `logs_cursors[cursor_index].status = DataCursor::ERROR;` in this case I think

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp [hunk 2]
```diff
@@ -1735,22 +1743,34 @@ bool ArbCore::handleLogsCursorRequested(ReadTransaction& tx,
...
+        logs_cursors[cursor_index].status = DataCursor::READY;
+        return;
+    }
+    if (current_count_result.data > log_inserted_count.data) {
+        // Error
+        std::cerr << "handleLogsCursor current count: "
+                  << current_count_result.data << " == "
+                  << " log inserted count: " << log_inserted_count.data
+                  << std::endl;
         logs_cursors[cursor_index].status = DataCursor::READY;
```
Comment (hkalodner) [inline packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1772]: ERROR rather than READY?

File: packages/arb-util/core/logreader.go [hunk 3]
```diff
@@ -101,73 +107,55 @@ func (lr *LogReader) getLogs(ctx context.Context) error {
 			time.Sleep(lr.sleepTime)
 		}
 
-		currentLogCount, err := lr.consumer.CurrentLogCount()
-		if err != nil {
-			return err
+		if len(logs) > 0 || len(deletedLogs) > 0 {
+			logger.Info().
```
Comment (hkalodner) [inline packages/arb-util/core/logreader.go:111]: I think we want to remove this

File: packages/arb-util/core/logreader.go [hunk 4]
```diff
@@ -101,73 +107,55 @@ func (lr *LogReader) getLogs(ctx context.Context) error {
...
+		if logs != nil || deletedLogs != nil {
+			logger.Info().Uint64("cursorIndex", lr.cursorIndex.Uint64()).Msg("confirming receipt of logs")
+			for {
+				status, err := lr.cursor.LogsCursorConfirmReceived(lr.cursorIndex)
+				if err != nil {
+					return err
+				}
+				if status {
+					// Successfully confirmed receipt of logs
+					logger.Info().Uint64("cursorIndex", lr.cursorIndex.Uint64()).Msg("confirmed receipt of logs")
```
Comment (hkalodner) [inline packages/arb-util/core/logreader.go:141]: Probably want to remove this

File: packages/arb-avm-cpp/data_storage/src/arbcore.cpp (hunk 1 above)
Comment (joshuacolvin0) [reply packages/arb-avm-cpp/data_storage/src/arbcore.cpp:1757]: sure

**Pull request merged**